### DISCUSSION TITLE
Introducing Clarion

### DISCUSSION THREAD

**Discussion by bytemaster** [bytemaster/clarion#2] [General]
### Introduction

Clarion aims to give everyone in the world the tools to broadcast their message to everyone who wants to hear their message without creating dependencies on centralized infrastructure. It will achieve this with a censorship resistant "friend to friend" network which will leverage the unused resources of your friends and family to distribute your content.

The ultimate goal is to provide the performance and reliability of a "centralized service" with the freedom and independence of a logically decentralized network. With the help of the Clarion community we can free our friends and family from the tyranny of Twitter, Facebook, YouTube, Amazon, Apple and Google and produce a social network free from manipulation and 3rd party dependence.

### Prior Art 

Clarion learns lessons from other projects, including [RetroShare](https://retroshare.cc), [Hive](https://hive.blog) (previously [SteemIt](https://steemit.com) ), and [Voice](voice.com). RetroShare is far closer to the desired level of decentralization, but has a clunky interface that isn't viable on mobile. Meanwhile, blockchain based networks are "logically centralized" and require all full nodes to process all transactions. This is not suitable for home users at scale and creates dependencies on a small number of web portals.  

One of the more successful decentralized protocols is IPFS and IPNS. These P2P protocols facilitate the publishing of location independent websites; however, they have a couple of drawbacks. Among these drawbacks are the need to keep your own node online in order to seed your content -- there is no free way to "push to IPFS". There is also a high latency to looking up content. Lastly, there is not really an opportunity for dynamic multi-user interaction.  

Protocols like [Zeronet](https://zeronet.io) provide an application framework to build dynamic websites which are loaded/shared over the BitTorrent protocol. My experience with Zeronet is that it doesn't work well on Mobile, it takes a long time to "sync" pages, and suffers from excessive spam or needed moderation. At scale, a Zeronet application is just a logically centralized as a blockchain application. 

### Progressive Web Application
One of the architectural design decisions that sets Clarion apart from most other peer to peer applications is the choice to utilize a progressive web application powered by Web Assembly. Recent actions by Google, Amazon, and Apple have demonstrated that we cannot rely on app stores and hosting providers to distribute our applications and content. This means the only viable way to get a censorship resistant application onto a cell phone is via Progressive Web Applications.

Web Assembly allows us to implement the logic of our social network in high-performance C++ while reusing most of the same code in the browser that we use in "full nodes" running on open platforms like desktops or personal servers. 

### Application Platform

The core of the Clarion OS is a friend-to-friend message propagation protocol that enables the following kinds of message flows between cryptographic identities:

1. asynchronous person to person (like email) 
2. realtime person to person (video/chat)
3. one to many (like tweet/Facebook)
4. many to many (like an open chatroom / blockchain ) 

Utilizing Web Assembly plugins it should be possible to build many different kinds of applications on top of Clarion OS, including tokens and smart contracts. A key distinction between Clarion OS and prior blockchain platforms is that the base layer has no need to reach consensus on the "order" of user actions nor to interpret the meaning of transactions. If typical Ethereum and EOS smart contracts are considered "layer 1", then Clarion OS could be considered "layer 0".   

### Project Status

Clarion OS is just getting started and is in the early design stage. We are assembling a team of developers to build the first prototypes. This project will be developed in the open and welcomes contributions from anyone interested in restoring the internet to the level of logical decentralization that was intended before Big Tech locked us into their services and stopped producing tools that empower people.

**Comment by kollab** [comment]
In a word:  Inspirational!

Over the coming years, this platform could have a vast & profound impact by delivering truly censor-resistant communication and data-sharing for Mankind.

Well done, sir!

**Comment by kryptokurrency** [comment]
Is this gonna be a SERVER-LESS app?

**Reply by bytemaster** [reply]
That is the intention, no central server necessary.

**Reply by kryptokurrency** [reply]
Which protocol do you have in mind?

TCP
UDP
UDP + TCP

**Reply by bytemaster** [reply]
All of the above and more.  The first layer will be WebSocket (wss), then we are looking a quic (udp) for the low-latency connections and NAT traversal.

**Reply by chadrickm** [reply]
> no central server necessary.

I think you did some research into The Safe Network (then Maidsafe) some years back and didn't like it for a few different reasons. I think it's quite different now and you might want to do another checkup on it to see if there is anything that might be applicable.

**Reply by theblockstalk** [reply]
perhaps WebRTC could be useful for this. We built a prototype where the WebRTC handshake brokering service was done pseudo-anonymously in a smart contract, it worked...

**Reply by vaultec81** [reply]
I think it would be good to use libp2p as it already supports UDP and TCP with a range of different protocols and encryption mechanisms on top of that.

**Reply by jjmcclure** [reply]
What is the best way for an investor who wants to support this endeavor to get involved with the project?

**Reply by compumatrix** [reply]
Looks promising.

**Reply by eikrlan** [reply]
Why not IPv6 instead of NAT traversal ?

**Reply by faddat** [reply]
kcp

@eikrlan the reason not ipv6 instead of nat traversal is that v6 is poorly deployed and poorly supported.

**Reply by faddat** [reply]
though, revisiting this post, it surely should be both.

**Comment by mrchenmo** [comment]
Three Family Dogs go！go！go！

**Comment by seansoe** [comment]
Congrats! Looking forward to its progress

**Comment by kcoder2016** [comment]
Will b1 support this project with funds? and will it do airdrop to every EOS holder?

**Reply by tpkeeper** [reply]
1

**Reply by bytemaster** [reply]
If there are tokens, then they could be airdropped on EOS.

**Comment by interwebcoding** [comment]
The previous era was built by railroad tycoons and their wealth.  

One of the exciting promises of #bitcoin was that it would create a new age of wealth in the hands of people with a combination of cypherpunk / anarchist / libertarian / austrian economics mentalities.  

This is the exact sort of project I would hope to see succeed and fight back against the tyranny  that we see in big government and big tech.

**Comment by miron** [comment]
If Clarion OS is considered layer 0, what layer is ICMP-Chat considered then, -1? :)
http://icmpchat.sourceforge.net/

**Reply by jonathanbahai** [reply]
Subzero

**Reply by RdeWilde** [reply]
@jonathanbahai damn, you still around? What a small world. Go Peerplays

**Comment by kcoder2016** [comment]
how many  distributed node servers it need, which store the messages?

**Reply by bytemaster** [reply]
It should work from a single server to any number of servers.

**Reply by kcoder2016** [reply]
It says that message no cache on server, the message trans by peer to peer, but how the friends get the message when I post 
 during they offline?

**Reply by Headieddie** [reply]
I would imagine it resides on the commenter's computer until that 2nd person comes back online and pull it to theirs. It could also be pulled from friends of the commenter who were online and backed it up.

**Reply by kcoder2016** [reply]
> I would imagine it resides on the commenter's computer until that 2nd person comes back online and pull it to theirs. It could also be pulled from friends of the commenter who were online and backed it up.

But this time they are all offline who have pulled message, how to do that

**Reply by Headieddie** [reply]
Over my head, but maybe one friend of each group, is a service that records but then auto deletes if a few others have grabbed it. Maybe that same "friend"/service could detect when too many parties are going offline, grab what they can from remaining nodes and store it for a predetermined time, and maybe then push to a blockchain at another set limit, prepaid.

**Comment by kcoder2016** [comment]
who can delete the message beyond the publisher & how to process the illegal message?

**Reply by bytemaster** [reply]
You can only request the deletion of the message on your computer and from your friends computers, but if anyone else gets a copy then you cannot delete it (you can't remove it from your followers feeds once they get it).

The only content on your computer will be from your friends or those you follow (or anyone they "retweet"). You can also trust your friends moderation to delete content you don't want that is published by someone you follow. No random people will be able to push content to you.

**Reply by diceangel20** [reply]
> You can only request the deletion of the message on your computer and from your friends computers, but if anyone else gets a copy then you cannot delete it (you can't remove it from your followers feeds once they get it).
> 
> The only content on your computer will be from your friends or those you follow (or anyone they "retweet"). You can also trust your friends moderation to delete content you don't want that is published by someone you follow. No random people will be able to push content to you.

Does it sound like multiple ipfs private networks type application?

**Comment by mrdoom4334** [comment]
Hey Daniel sorry i have no knowledge of programming and i want to know is Clarion an ecosystem of its own like a new blockchain or it will be build on top of a dapp that runs on EOS or eth

**Reply by bytemaster** [reply]
Blockchains will run on top of ClarionOS not the other way around.

**Reply by Mythocurrency** [reply]
Cool, cool! Sounds like an operating system for the entire planet Earth.

**Comment by bdbstl** [comment]
Will there be a token for monetization?

**Reply by bytemaster** [reply]
The platform does not have a token at its foundation. It can integrate with existing token systems.

**Reply by eoswatchdogs** [reply]
Welcome to integrate with the vigor stablecoin protocol. Give users a better way to hodl their tokens.

**Comment by Cannyge** [comment]
When to issue the token?

**Reply by Mythocurrency** [reply]
Please. No token.

**Comment by mrdoom4334** [comment]
Another question... what will happen to harmful/bad content being broadcasted on Clarion How will it be solved / dealt with ...and who will have such authority to do such thing

**Reply by bytemaster** [reply]
People stop following/friending them.  Content only ever follows intentional connections. For example, I would never see a "your a scammer" message.

**Reply by cifilter** [reply]
> People stop following/friending them.  Content only ever follows intentional connections. For example, I would never see a "your a scammer" message.

This is the correct answer. "Bad/Harmful content" is arbitrary, and society should directly police what they deem to be good or bad content strictly by choosing to consume or not consume that content. As soon as some governing authority gets to make these decisions for everyone, then this platform would no longer truly be decentralized.

**Reply by Mythocurrency** [reply]
So should I be able to plagiarize content you produce on Clarion and take credit for it as my own? "Yes, I am stealing from you and passing it off as my original work. If you don't like it, go away!"

If the answer is yes, then there should absolutely be 100% KYC.

**Reply by O1O1O1O** [reply]
> Another question... what will happen to harmful/bad content being broadcasted on Clarion How will it be solved / dealt with ...and who will have such authority to do such thing

You're thinking of centralized services like Facebook and Twitter who host and broadcast content. Messages flowing on a service like Clarion are more like someone sending you a text - except the text cannot be read by the phone company. If you receive objectionable texts you block their number which is equivalent of not being their friend any more.

**Comment by bytemaster** [comment]
A lot of the concepts here are similar to https://holochain.org, leaving a note here to investigate further what lessons can be learned from holochain or if we can leverage their work.

**Reply by iamveritas** [reply]
If you are looking at holochain then nano I think it's worth looking at as well.

**Reply by pdaoust** [reply]
Hi, Holochain core team member here. Happy to talk if you have questions.

I'd also encourage you to check out Secure Scuttlebutt https://scuttlebutt.nz , which is similar to Holochain but uses the friend-to-friend propagation model you're describing. Hypercore https://hypercore-protocol.org is similar too, but uses a more BitTorrent- or IPFS-like propagation model (that is, I seed the things I'm interested in, and a DHT handles seeder discovery).

**Reply by pegaltier** [reply]
I'm glad to see this discussion. @bytemaster you should consider joining the holochain community. They would be happy to work with you.

**Reply by O1O1O1O** [reply]
It seems like utilizing Holochain as a host for data solves the problem of "I'm on a new computers and all my friends are offline, how do I get their messages" or "my friends sent me a message but his browser is offline now so how to I fetch it".

**Reply by faddat** [reply]
All the smartest people from rchain headed to holochain

**Comment by rd123myb** [comment]
Hi Dan, I'm just wondering if Clarion will port data from existing social media networks? I believe the data governance act article 20 may soon require by law that social platforms give users the ability to authorize access to their whole history via an API. There was a discussion about this talking about this by founders of some other data related projects. https://youtu.be/fJ42u4IiVTM?t=2480

**Reply by bytemaster** [reply]
This will certainly be a possibility worth looking into.

**Comment by ghost** [comment]
Right on! As they say, "In Layer 0 was the Word, and the Word was with Freedom, and the Word was Freedom."

**Comment by elmato** [comment]
Hi Dan, Clarion will be something similar to https://safenetwork.tech/ (MaidSafe) ?

**Reply by bytemaster** [reply]
I'm leaning against a DHT. After 12 years maidsafe has yet to produce something I can use.

**Comment by xyuger** [comment]
Will it integrate with eos.io systems?

**Comment by Velua** [comment]
Would [LibP2P](https://libp2p.io/) be of any use here? It will be a damn shame if there's more reinventing the wheel than required given all the efforts already made in several Decentralisation efforts.

**Reply by bytemaster** [reply]
I've looked at libp2p. I will have to provide a more extensive discussion on pros and cons.

**Comment by sportland1968** [comment]
Great news, sir!

**Comment by yilunzhang** [comment]
Have you considered leveraging existing decentralized p2p protocol like https://github.com/nknorg/nkn-sdk-go as your networking layer? Sounds like a good fit: decentralized, layer-0, websocket, etc.

**Reply by bytemaster** [reply]
Thanks for sharing this link. I will take a look.

**Comment by dosper7** [comment]
Why not supporting other type of more popular languages like csharp, java, python, etc In order to attract more developers to this?

**Reply by bytemaster** [reply]
I'm targeting wasm because of need for smart contract like determinism and c++ for performance.  I suppose Rust is an option.

**Reply by O1O1O1O** [reply]
Indeed when you say Wasm now you're basically saying "it runs on the web/internet" some that is it's equivalent to machine code. There are many languages targeting Wasm now since many languages target LLVM and LLVM can target Wasm. Everything else is down to libraries.

**Comment by smsbyy** [comment]
Only  Looking forward to this new project !

**Comment by samkirton** [comment]
Have you considered Rust as the primary language for the project?

**Reply by bytemaster** [reply]
I'm not an expert in Rust nor is the team I have assembled.

**Reply by samkirton** [reply]
aside from low level networking, crypto, utils etc., the system could be comprised of "plugins" which accept a WASM module conforming to an export function interface, it would be nice to be able to collaborate in a variety of languages that compile into WASM

**Comment by Jeff-Hang** [comment]
Will it be launched on the EOS mainnet?

**Reply by bytemaster** [reply]
No

**Comment by iamveritas** [comment]
I can't wait to see what it will improve over existing ideas and implementations.
Godspeed.

**Comment by UrsaPolarisRecords** [comment]
The one thing that worries me a little bit is how to prevent this solution from further amplifying echo chambers. Some method of mixing in humanized different opinions would be wise, in my view. Something like showing posts from people you may know, or posts that your friends like. Just a thought I had.

**Reply by bytemaster** [reply]
You would see things your friends like.

**Reply by Headieddie** [reply]
Yes, but if the messages are still viewable to all,  others can make a comment that references another comment, regardless of original poster's will to see or store the data. So Dan wouldn't see the scammer comments unless he turned on the layer that showed those comments. There could then be rating systems so that he could leave that layer off at all times, but be notified of high ranking/engaging comments,  and can then have it pushed into his viewer or have a look manually. "Pushed" would not mean storing the data himself.

**Reply by bytemaster** [reply]
Actually, my computer would never see the "scammer" comments unless someone I follow interacted with them in a positive way. People wouldn't be able to leverage the visibility of my large following to distribute their hate.

**Reply by dallasjohnson** [reply]
I can't see how this would not lead to an amplification of in-group biases. Then this would lead to social censorship from your friends and family and more confirmation biases of a person's existing beliefs without ever being uncomfortably challenged.

**Reply by bytemaster** [reply]
It is not any platforms job to make you see stuff that you don't want. Each person must choose to seek out info and use software in ways that expose them to stuff. 

The whole point is to take power from platforms and give it to users.

**Comment by jawadhaxan** [comment]
P2P, a server-less network, with high-performance web-assembly PWA, was thinking of something similar.
I am worried about the misuse of this censorship-resistant app, which should be avoided by Decentralized governance, in my view.

**Reply by O1O1O1O** [reply]
You might want to read Dan's book More Equal Animals.

**Comment by On1x** [comment]
Hello there, Dan!

I'm thinking about same concept and working on it like 1 year :)

Starting with similar ideas — user subscribes on each other and load from block log only neccesary information with custom protocol.
Awards working with social capital VIZ: https://github.com/vizplus/viz-faq/blob/main/faq-en.md
PWA that usign IPFS/SIA for images, and public gates for accounts metadata and activity. Social network algoritm on the client side.

Each account got custom operation last activity sequence, that way anyone can build activity chain for specified user.
![l2Nxc3c](https://user-images.githubusercontent.com/1928273/109798692-836c7980-7c34-11eb-82ef-48da99e00499.png)

Articles about concept:
- [The protocol of a decentralized social network on the distributed ledger](https://control.viz.world/media/@on1x/en-decentralized-social-network-protocol-on-blockchain/)
- [Building a new protocol on the VIZ blockchain](https://control.viz.world/media/@on1x/en-custom-protocol-on-viz-blockchain/)
- [The Voice protocol on the VIZ blockchain](https://control.viz.world/media/@on1x/en-protocol-voice-on-viz-blockchain/)
- [Use of an intermediate hub in the Voice protocol](https://control.viz.world/media/@on1x/en-using-proxy-hub-protocol-voice/)

I'm develop The Free Speech Project and start it with https://Readdle.me/ 👍 
https://control.viz.world/media/@on1x/en-readdle-me-beta/

There is lack of third-party services, but it can be resolved by microservices with passwordless access: https://readdle.me/#dapp:manual/services/

Will be happy with feedback from community!
![iCQ4aeL](https://user-images.githubusercontent.com/1928273/109798760-9717e000-7c34-11eb-8f20-9e0ad3f175a6.png)

**Comment by lukestokes** [comment]
Someone asked me to ask you about https://urbit.org/. Is it similar? I hadn’t heard of it before being asked.

Is it possible to expand the prior art section extensively before committing to building something new? I understand how enjoyable it is to have a green field to explore new ideas unconstrained, but I also think there may be great projects out there which could benefit from your experience and involvement. What if we all started building more interoperability between existing platforms and protocols instead of starting over each time?

Either way, I hope you succeed in your stated intentions of building tools for freedom and liberty.

I wonder how much this will piss off B1 as it seems like an eventual competitor to Voice. I hope there were no non-compete clauses in your previous employment contract.

**Reply by bytemaster** [reply]
I'm considering all existing art and would love to compare/contrast with what is out there. That's why I announced this because it's difficult to search for everything and far more productive for people to share what they know.  

Thanks for the referral, I'll look into it.

**Reply by pdaoust** [reply]
Yeah, I see that Urbit could be worth checking out too. I can't say I totally understand it, and I know that there are questions about the ethics of its creator Curtis Yarvin (AKA Mencius Moldbug), but a former colleague of mine who now works at Urbit tells me they've jettisoned a lot of that baggage associated with him. From what I can tell, it's an OS, a low-level machine language, a high-level programming language, a 'true' (logically decentralised) P2P protocol, and a scarce namespace secured by blockchain.

**Comment by Dexaran** [comment]
>A key distinction between Clarion OS and prior blockchain platforms is that the base layer has no need to reach consensus on the "order" of user actions nor to interpret the meaning of transactions

Does this actually mean that user actions do not have timestamp (or have variable timestamp which can be manipulated to some degree)?

>Utilizing Web Assembly plugins it should be possible to build many different kinds of applications on top of Clarion OS, including tokens and smart contracts.

There are two problems of smart-contracts across the industry:

1. The lack of built-in source of entropy
2. The lack of continuous processes and possibility of scheduling actions in the future (i.e. a server can "do something once per 10 minutes" while smart-contracts can only do something upon invocation). 
[Deferred transactions](https://github.com/EOSIO/eos/releases/tag/v2.0.0-rc1) were deprecated in EOS.

Will these problems be addressed with Clarion contracts?

**Reply by jbuck84** [reply]
A couple of great questions for @bytemaster right here!

**Reply by bytemaster** [reply]
All user actions have a timestamp that must be less than "now" in order to propagate across the network.

No solution to the entropy problem.

No ClarionOS level consensus algorithm, but nothing stops "deferred transactions" from working at higher levels.

**Comment by ismyhc** [comment]
Good morning everyone. I think this is a noble endeavor. Since the recent happenings of censorship going on in the world, I had a feeling that your departure from b1 would result in you taking this journey. I welcome this for sure and am very interested in what you are doing. I think it's quite clear now that services and people can not only be cut of at the platform level, but also at the service provider level and even ISP level. 

That said, I think one thing that should be focused on from the very beginning with ClarionOS is ease of use and simplicity on the frontend side. This was one of the things I think that b1 missed the boat on, but we at Lynx picked up on early. If ease of use can be kept in mind at the beginning I think it will make a lot of difference in adoption.

I look forward seeing this grow and hope to contribute where I can.

**Reply by bytemaster** [reply]
I agree about usability of front end being top priority.

**Reply by lukestokes** [reply]
I hope the [FIO Protocol](https://fioprotocol.io/) can help with this as well. We're an EOSIO-based chain providing human readable addresses, decentralized payment requests, and meta data about transactions. Works with any chain or token and is already integrated into over a dozen wallets and exchanges. I hope whatever identity system Clarion uses will also work well with FIO. I'd love to help however I can in regards to usability as I also agree it's a huge priority. As we [transition out of innovators to early adopters](https://docs.google.com/presentation/d/1ZOj2n13yt5BSH0o7kNjQ1hWTlEK2yEJdjWX6ILci7Ec/edit?usp=sharing) and if the exponential growth of crypto continues, the vast majority of users in crypto that are just around the corner will require a much better user experience.

**Comment by mapodev** [comment]
This is releasing my heart. Freedom!

**Comment by Ashrith333** [comment]
would love to contribute and see this project grow , goodluck !

**Reply by bytemaster** [reply]
Thanks, what ways do you think you could contribute?

**Comment by dskvr** [comment]
> The ultimate goal is to provide the performance and reliability of a "centralized service" with the freedom and independence of a logically decentralized network. With the help of the Clarion community we can free our friends and family from the tyranny of Twitter, Facebook, YouTube, Amazon, Apple and Google and produce a social network free from manipulation and 3rd party dependence.

This sounds a lot like [Scuttlebutt Protocol](https://en.wikipedia.org/wiki/Secure_Scuttlebutt), have you checked it out?

**Reply by lukestokes** [reply]
Another one to add to the Prior Art section.

**Reply by bytemaster** [reply]
See the critique of Scuttlebutt Protocol provided by https://github.com/fiatjaf/nostr  which is the most similar project I have seen so far.

**Reply by dskvr** [reply]
As a user of SSB for a while now I agree with all those critiques, particularly this one:
> It insists on having a chain of updates from a single user, which feels unnecessary to me and something that adds bloat and rigidity to the thing — each server/user needs to store all the chain of posts to be sure the new one is valid. Why? (Maybe they have a good reason);

**Comment by pravvy** [comment]
Good luck from your friends at EOS Miami

**Comment by adsorptionenthalpy** [comment]
I am very much looking forward to this project's progress. Great stuff!

**Comment by alexliqq** [comment]
I have been trading bts/steem/eos tokens and participate in these community and building for year, as token holders of eos/bts and community members I would like to ask: do you have suggestions to community members on how to participate in Clarion, will there be any kind of token sales or crowdfunding and how to incentive developers to build on Clarion OS.

**Comment by juicebyjustin** [comment]
I'm a .net developer looking to expand my skill set and knowledge to blockchain. Will there be an open call for developers or do you have a team assembled?

**Reply by bytemaster** [reply]
We will be opening this up for many volunteers to participate. I hope to create a culture around mutual sharing and team effort instead of an "insiders" group.

**Comment by mynameisek** [comment]
friends of mine working on this protocol,
might be very interesting to partner with:

https://berty.tech/

**Reply by bytemaster** [reply]
Thanks for the link. Very fascinating. I looked at it and tried it out. Doesn't accomplish my goals.

**Comment by momack2** [comment]
Heyo - IPFS person here -- excited to hear you want to build on IPFS/IPNS! There are already a few different strategies for nodes to persist data on the IPFS network (see https://filecoin.io, https://pinata.cloud/, and https://collab.ipfscluster.io/ for ways to ensure data is available when your node goes offline) - but a friends-of-friends replication layer sounds like an exciting addition to the option space!

For your point about dynamic multi-user apps - we've seen an increase in these over the past year, but agree there's huge opportunity for a firebase-style application framework to build dynamic websites more easily. Groups like @3box @infinitescroll and @unstoppabledomains have experimented with this too! There's also already a c++ libp2p implementation and many of the ipfs components from the soramitsu team to build on. =] Would love to find ways to support you!

**Reply by bytemaster** [reply]
Welcome!  I absolutely believe there is room to collaborate between these projects. I've spent time testing ipfs and libp2p.

**Reply by dschlossman** [reply]
This project reminds me of the diaspora project and its 3 main goals: decentralization, freedom, and privacy --> https://diasporafoundation.org/

**Reply by bytemaster** [reply]
@dschlossman thanks for the link to diaspora. I looked deeper into it and discovered WebSub which is the underlying protocol. This is what I would call a Federated model where "normal" users are clients of a particular hub and the hub is a server. It is not clear that user accounts can easily migrate from hub to hub (even though they can communicate from hub to hub).

**Reply by O1O1O1O** [reply]
I've played with Diaspora as a user and it didn't work for me.  Having to know what hub someone is connected to just seems weird.  It's like the old UUCP based email where you had to know a path between all the connected servers to reach someone.  Reaching me should be independent of what many-to-many communities I'm involved with.

**Comment by TendieShop** [comment]
Looking forward to seeing what you will build!

**Comment by qimiaoguo** [comment]
I like the tech byte master

**Comment by Avm07** [comment]
Great idea,I would be happy to join the dev team or contribute to this project

**Reply by bytemaster** [reply]
What kind of skills do you have?  Any other projects that you are involved with?

**Reply by Avm07** [reply]
C++ 17(STL,Boost,CMake), EOS smart contracts

Last 2 years i focus on EOS blockchain

- developing smart contracts, have experience with plugins
- writing micro services

Main project: https://paycash.app
DeFi already launched on EOS mainnet, in closed beta.

Last project: https://github.com/Avm07/mindswap-smart-contracts
Automative market making for EveripediaNetwork, simillar to Uniswap.

There are a lot of more dApps which is currently launched on EOS mainnet, but not open sourced.

**Comment by atriantafy** [comment]
Any plans to add Python bindings? I think that would make it more accessible to the developers community and allow for more applications to flourish on top of it.

**Reply by ghost** [reply]
I think it would be great. Python is fascinating.

**Comment by ancjf** [comment]
I have a similar idea. You should do better than me！

**Comment by atriantafy** [comment]
On the issue of sensitive conten: how about Deep Learning components that run on the network, and perform some filtering? For example, there could be a porn filter. It would be then up to the user to decide whether they'd like to turn the filter on / off depending on their use-case. Using bounties, deep learning engineers could continue improving the algorithms.

**Reply by Velua** [reply]
Isn't the user only gonna see content of people he's following?

**Comment by O1O1O1O** [comment]
I assume there is no formal kind of identity system other than perhaps signing of a message so messages signed with that same key are assumed to come from the same person (or joint holders of that key).

Is the only means for discovery by a person to person invite - scan a QR code to add someone as a friend or get one via an existing third party system. But once connected to at least one person I could not discover how to add say Dan Larimer unless someone I know shares a message from him. And even then I could not be sure which Dan Larimer it is. Is that correct?

**Comment by O1O1O1O** [comment]
Do my friends get a copy of all of my messages or just the ones I've sent/shared with them?   For instance I have friends Bob and Alice and send Bob a message from a browser on my home computer.  Now I got to the library and get online from a new browser - it obviously doesn't have any Clarion state and my home computer is off.  If Bob's computer is now offline will Alice have a copy (encrypted) of the message I sent Bob?  Or I can I now only see the union of all messages that were sent by me to people with browsers currently online?

Actually can I even access Clarion as me if all my other browsers that have "me" as state in them are offline?  Or does it always require a sync from an existing system to know who all my friends are?

**Comment by 7flash** [comment]
:eyes:

**Reply by 7flash** [reply]
I am happy to contribute too. Successfully launched EOS Smart contracts for Regenerative Economy - github.com/JoinSeeds -. Safe and Free Social Network is next evolutionary step towards truly Regenerative Society.

**Comment by amih** [comment]
Sounds like the enabling tech for the shower coin, or is it the new incarnation for that idea?
Are there any ideas for hardware networking layers to make this independent of the big tech and making it more resistant to being interrupted by outsider companies?
Perhaps EdenOS communities can implement an Infra-red network of servers which will be the backbone of the clarion network for the community. (Just a half baked idea)

**Comment by gorakhborude** [comment]
I am developer and I am willing to contribute to this project. I am aware of C# , Java , node js, asynchronous messaging , react js and cryptography technics.

**Comment by keithofaptos** [comment]
Any way to integrate cMIX, dVPN, and dClouds into ClarionOS? For true E2E. As you're likely very aware that without encryption there is zero digital freedom. Plus quantum processing is here, now. And RSA's days are numbered. PS- I dig how you think. And am available to assist.

**Comment by ackza** [comment]
Yes the hardware is you the wallet is YOU

Quilts and pushcards Linda love lace

**Comment by LasseEhlers** [comment]
This is a much needed product in the world of today, censorship already is ramped, I believe Daniel Larimer can lead the team to accomplish decentralized communication like this!

**Comment by grepsuzette** [comment]
Imagine a tech that would eliminate the need for email and phone altogether. 
And that tech is not proprietary. 
That would be a worthwhile advancement.

**Comment by LeonAllen** [comment]
This is weicheng你发给我的信件已经收到。This is an automatic reply, confirming that your e-mail was received.Thank you
